File: python/217 - Contains Duplicate.py

class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        setNum = set()

        for num in nums:
            if nums in setNum:
                return True
            setNum.add(num)
        return False


# Time limit for complexity

# class Solution:
#     def containsDuplicate(self, nums: list[int]) -> bool:

#         for i in range(len(nums)):
#             for j in range(i+1, len(nums)):
#                 if nums[i] == nums[j]:
#                     return True
#         return False

File: TypeScript/src/217_ContainsDuplicate.ts
function containsDuplicate(nums: number[]): boolean {
    const sortsNums = new Set<number>();

    for (let num of nums) {
      if (sortsNums.has(num)){
            return true;
      }
      sortsNums.add(num);
    }
    return false;

};




/***
 * this code is time limit 
 * O(n^2) run time 
 * 
 *  ***/

// function containsDuplicate(nums: number[]): boolean {

//     for (let i = 0; i < nums.length; i++) {
//         for (let j = i + 1; j < nums.length; j++) {
//             if (nums[i] == nums[j]) {
//                 return true;
//             }
//         }
//     }

//     return false;

// };

